githubgen/codeowners.go

```diff
@@ -32,73 +36,100 @@ func (cg *codeownersGenerator) Generate(data datatype.GithubData) error {
-				if !strings.HasPrefix(owner, "@") {
-					owners += "@" + owner
-				}
+				owners += formatGithubUser(owner)
 			}
-			codeowners += fmt.Sprintf("%s/%s %s%s\n", strings.TrimPrefix(folder, data.RootFolder+"/"), strings.Repeat(" ", data.MaxLength-len(folder)), data.DefaultCodeOwner, owners)
+			ownerComponents = append(ownerComponents, fmt.Sprintf("%s/%s %s%s", strings.TrimPrefix(folder, data.RootFolder+"/"), strings.Repeat(" ", data.MaxLength-len(folder)), data.DefaultCodeOwner, owners))
 		}
 	}
 
-	codeowners += distributionCodeownersHeader
 	longestName := cg.longestNameSpaces(data)
 
 	for _, dist := range data.Distributions {
 		var maintainers []string
 		for _, m := range dist.Maintainers {
-			maintainers = append(maintainers, fmt.Sprintf("@%s", m))
+			maintainers = append(maintainers, formatGithubUser(m))
 		}
 
-		distribution := fmt.Sprintf("\nreports/distributions/%s.yaml%s %s", dist.Name, strings.Repeat(" ", longestName-len(dist.Name)), data.DefaultCodeOwner)
+		distribution := fmt.Sprintf("reports/distributions/%s.yaml%s %s", dist.Name, strings.Repeat(" ", longestName-len(dist.Name)), data.DefaultCodeOwner)
 		if len(maintainers) > 0 {
 			distribution += fmt.Sprintf(" %s", strings.Join(maintainers, " "))
 		}
 
-		codeowners += distribution
+		distributions = append(distributions, distribution)
+	}
+
+	// CODEOWNERS file
+	codeownersFile := filepath.Join(data.RootFolder, ".github", "CODEOWNERS")
+	templateContents, err := cg.getFile(codeownersFile)
+	if err != nil {
+		return err
 	}
 
-	err = os.WriteFile(filepath.Join(data.RootFolder, ".github", "CODEOWNERS"), []byte(codeowners+unmaintainedCodeowners), 0o600)
+	templateContents = injectContent(startComponentList, endComponentList, templateContents, ownerComponents)
+	templateContents = injectContent(startDistributionList, endDistributionList, templateContents, distributions)
+	templateContents = injectContent(startUnmaintainedList, endUnmaintainedList, templateContents, unmaintainedCodeowners)
+
+	err = cg.setFile(codeownersFile, templateContents)
 	if err != nil {
 		return err
 	}
-	err = os.WriteFile(filepath.Join(data.RootFolder, ".github", "ALLOWLIST"), []byte(allowlistHeader+deprecatedList+unmaintainedList), 0o600)
+
+	// ALLOWLIST file
+	allowListFile := filepath.Join(data.RootFolder, ".github", "ALLOWLIST")
+	allowListContents, err := cg.getFile(allowListFile)
+	if err != nil {
+		return err
+	}
+
+	allowListContents = injectContent(startUnmaintainedList, endUnmaintainedList, allowListContents, allowListUnmaintainedComponents)
+	allowListContents = injectContent(startDeprecatedList, endDeprecatedList, allowListContents, allowListDeprecatedComponents)
+
+	err = cg.setFile(allowListFile, allowListContents)
 	if err != nil {
 		return err
 	}
 	return nil
 }
 
+func injectContent(startMagicString, endMagicString string, templateContents []byte, replaceContent []string) []byte {
+	matchOldContent := regexp.MustCompile("(?s)" + startMagicString + ".*" + endMagicString)
```

Comment:
are we sure it will always compile? I would rather use `regexp.Compile` and check for the error in case the regex string is invalid

Reply:
i think it is going to always compile. the regular expression is made up of all constant parts so there should be no variation at all.